### PR TITLE
Remove some uses of `unwrap()`

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

**Comment by sgrif** [comment]
Can you pick one style of map(collect) and use it consistently? This looks fine otherwise

**Comment by sgrif** [comment]
Also you shouldn't need to manually use `Iterator`. It's in prelude.

**Comment by tamird** [comment]
Done and done.

File: diesel/src/expression/expression_methods/global_expression_methods.rs [hunk 1]
```diff
@@ -295,11 +295,10 @@ pub trait ExpressionMethods: Expression + Sized {
     ///     use self::posts::dsl::{posts, author_name};
     ///     let connection = establish_connection();
     ///
-    ///     let data = users.inner_join(posts)
+    ///     let data: QueryResult<Vec<String>> = users.inner_join(posts)
     ///         .filter(name.nullable().eq(author_name))
```
Comment (mcasper) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: We should probably decide on style guidelines for when chaining functions together across multiple rows. One function per row? As many as will fit within the soft 80 char limit?

Comment (mfpiccolo) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: One function per line is much easier to parse than:

```
let data: QueryResult<Vec<String>> = users.inner_join(posts)
    .filter(name.nullable().eq(author_name)).select(name).load(&connection)
    .map(Iterator::collect);
```

Did you add the comment to this line to address the filter() function specifically?

I don't think we need to separate this to be multiline though:
```
.filter(
    name.nullable()
        .eq(author_name)
)
```

Comment (mcasper) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Not specifically the `filter` line, just that this chain demonstrated the mixing of styles.

So maybe the guideline is to keep each line of the chain short and readable then
```Rust
let result = my_struct.foo_function()
  .bar_function().unwrap()
  .baz_with_longer_sig()
```

Comment (sgrif) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Generally one per row

Comment (sgrif) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Things that are likely to change for the same reason can be grouped (`.unwrap().collect()` for example)

Comment (tamird) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: To clarify, are you asking that I make that change throughout?

Comment (sgrif) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Yes, please

Comment (tamird) [inline diesel/src/expression/expression_methods/global_expression_methods.rs]: Done.

**Pull request merged**